libs/sysplugins/smarty_internal_method_loadplugin.php

```diff
@@ -84,7 +84,7 @@ public function loadPlugin(Smarty $smarty, $plugin_name, $check)
             // loop through plugin dirs and find the plugin
             foreach ($_p_dirs as $_plugin_dir) {
                 foreach ($_file_names as $name) {
-                    $file = $_plugin_dir . $name;
+                    $file = $_plugin_dir . strtolower($name);
```

Comment:
The code surrounding this change has explicit checks for lowercasing. This change would break when loading a plugin with a case sensitive name on case sensitive filesystems.